volume/store/store.go

```diff
@@ -302,16 +302,14 @@ func (s *VolumeStore) Dereference(v volume.Volume, ref string) {
 
 	s.globalLock.Lock()
 	defer s.globalLock.Unlock()
-	refs, exists := s.refs[v.Name()]
-	if !exists {
-		return
-	}
```

Comment:
We aren't tracking existence anywhere as it's not what matters. What matters here is length (ie, 0 length it has no references, > 0 length it has references).